gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -790,14 +788,19 @@ _message_handler (void *thread_data)
     }
 
     if (result || gst_caps_can_intersect (client_caps, server_caps)) {
+      free (cmd_data.data.data);
       cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_APPROVE;
       cmd_data.data.data = (uint8_t *) _sdata->sink_caps_str;
       cmd_data.data.size = (size_t) strlen (_sdata->sink_caps_str) + 1;
     } else {
       /* respond deny with src caps string */
+      gchar *str = gst_caps_to_string (client_caps);
       nns_loge ("Query caps is not acceptable!");
-      nns_loge ("Query client sink caps: %s", (char *) cmd_data.data.data);
+      nns_loge ("Query client sink caps: %s", str);
```

Comment:
In my opinion, this is an duplicated works.
You can just use 'cmd_data.data.data' because you changed it to release ` cmd_data.data.data` after use.

Reply:
Thank you! I will fix it :)